Tests/Foundation/Tests/TestAttributedStringSupport.swift

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if NS_FOUNDATION_ALLOWS_TESTABLE_IMPORT
+    #if canImport(SwiftFoundation) && !DEPLOYMENT_RUNTIME_OBJC
+        @testable import SwiftFoundation
+    #else
+        @testable import Foundation
+    #endif
+#endif
+
+extension NSAttributedString.Key {
+    static let testInt = NSAttributedString.Key("TestInt")
+    static let testString = NSAttributedString.Key("TestString")
+    static let testDouble = NSAttributedString.Key("TestDouble")
+    static let testBool = NSAttributedString.Key("TestBool")
+    static let link: NSAttributedString.Key = .init(rawValue: "NSLink")
+}
+
+struct Color : Hashable, Codable {
+    let name: String
+    
+    static let black: Color = Color(name: "black")
+    static let blue: Color = Color(name: "black")
+    static let white: Color = Color(name: "white")
+}
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+extension AttributeScopes.TestAttributes {
+
+    enum TestIntAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = Int
+        static let name = "TestInt"
+    }
+
+    enum TestStringAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = String
+        static let name = "TestString"
+    }
+
+    enum TestDoubleAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = Double
+        static let name = "TestDouble"
+    }
+
+    enum TestBoolAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = Bool
+        static let name = "TestBool"
+    }
+    
+    enum TestForegroundColorAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = Color
+        static let name = "ForgroundColor"
+    }
+    
+    enum TestBackgroundColorAttribute: CodableAttributedStringKey, MarkdownDecodableAttributedStringKey {
+        typealias Value = Color
+        static let name = "BackgroundColor"
+    }
+
+    #if false
+    @frozen enum MisspelledAttribute : CodableAttributedStringKey, FixableAttribute {
```

Comment:
We should not have this if we're not compiling this, and Ortography isn't supported by swift-corelibs-foundation.